nst/models/baseline.py

```diff
@@ -74,19 +76,31 @@ def __init__(
 
         self._optimized_image = content_image
 
+        self._content_weight = content_weight
+        self._style_weight = style_weight
+        self._total_variation_weight = total_variation_weight
+
         self._learning_rate = learning_rate
-        pass
 
     def training_step(self, batch, batch_idx):
         content_feature_maps, style_feature_maps = self._feature_extractor(self._optimized_image)
 
         content_loss = self._content_loss(content_feature_maps)
         style_loss = self._style_loss(style_feature_maps)
-        total_variation_loss = self._total_variation_loss(self._optimized_image)
 
-        sum_of_losses = self.content_weight * content_loss + self.style_weight * style_loss
-
-        # TODO log losses and _optimized_image to tensorboard
+        total_variation_loss = self._total_variation_loss(self._optimized_image)
+        sum_of_losses = self._content_weight * content_loss + self._style_weight * style_loss
+
+        writer = SummaryWriter()
+        writer.add_image("Working Image", self._optimized_image.squeeze(), global_step=self.global_step)
+        writer.add_scalar("Loss/content/", content_loss, global_step=self.global_step)
+        writer.add_scalar("Loss/style/", style_loss, global_step=self.global_step)
+        writer.add_scalar("Loss/total_variation/", total_variation_loss, global_step=self.global_step)
+        writer.close()
+
+        # Open Tensorboard with the command:
+        # tensorboard --logdir=runs
+        # the runs folder is generated by default in the same path as the python script (here it would be nst/models)
```

Comment:
I'd remove this comment or move it to README to keep the code clear :)

Reply:
That's fair enough :) I shall move it now